### PR TITLE
added sigmoid bijectors on updated code

### PR DESCRIPTION
**Pull request by pabloswfly**

### CODE REVIEW

File: mcmcgan.py [hunk 1]
```diff
@@ -38,57 +38,66 @@ def D(self, x, num_reps=32):
...
 
-        score = tf.convert_to_tensor(self.D(proposals), tf.float32)
+        score = tf.cast(self.D(proposals), tf.float32)
+        print(tf.math.log(score))
         return tf.math.log(score)
 
-    def unnormalized_log_prob(self, x):
-        return tf.py_function(self._unnormalized_log_prob, inp=[x], Tout=tf.float32)
+    def target_log_prob(self, *x):
+        return tf.py_function(self._target_log_prob, inp=[x], Tout=tf.float32)
```
Comment (grahamgower) [inline mcmcgan.py]:
```suggestion
        return tf.py_function(self._target_log_prob, inp=x, Tout=tf.float32)
```

File: mcmcgan.py [hunk 2]
```diff
@@ -38,57 +38,66 @@ def D(self, x, num_reps=32):
...
         i = 0
-        x = x.numpy()
+        tf.print(x)
         for p in proposals:
             if proposals[p].inferable:
-                if not (proposals[p].bounds[0] < x[i] < proposals[p].bounds[1]):
-                    # We reject these parameter values by returning probability 0.
-                    return -np.inf
-                proposals[p].val = x[i]
+                proposals[p].val = x[0][i].numpy()
```
Comment (grahamgower) [inline mcmcgan.py]:
```suggestion
                proposals[p].val = x[i].numpy()
```

**Review by grahamgower — COMMENTED** [review]
I made the changes suggested below, which I think are appropriate. But then another problem arises: the `_target_log_prob()` function recieves lots of `-nan` values for the parameters. Not sure what's going on there.

**Comment by pabloswfly** [comment]
Super, thanks! I'll dig into what's happening.

File: mcmcgan.py [hunk 3]
```diff
@@ -38,57 +38,66 @@ def D(self, x, num_reps=32):
...
-            mcmc = tfp.mcmc.HamiltonianMonteCarlo(
-                target_log_prob_fn=self.unnormalized_log_prob,
-                num_leapfrog_steps=10,
-                step_size=self.step_sizes,
-            )
+            mcmc = tfp.mcmc.TransformedTransitionKernel(
+                tfp.mcmc.HamiltonianMonteCarlo(
+                    target_log_prob_fn=self.target_log_prob,
+                    num_leapfrog_steps=6,
+                    step_size=self.step_size,
```
Comment (grahamgower) [inline mcmcgan.py]: Hard-coding the step_size here seems to do the trick!
```suggestion
                    step_size=0.05,
```

**Comment by pabloswfly** [comment]
Cool, found it! It looks like, once the mcmc kernel is transformed with a bijector, the initial chain state is given in parameter values space, but the step size is given in pre-transformed space. Weird! The burn-in stage will do the magic.

**Comment by pabloswfly** [comment]
The sigmoid bijector works perfectly to map values to the parameter range of interest.

I also re-structured little parts of the code to make them simpler.

**Review by grahamgower — APPROVED** [review]
Looks good to me!

FYI, you're welcome to merge things without my explicit approval.

**Pull request merged**